src/cmap/connection.ts

```diff
@@ -165,6 +168,15 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
   public delayedTimeoutId: NodeJS.Timeout | null = null;
   public generation: number;
   public readonly description: Readonly<StreamDescription>;
+  /**
+   * @public
+   * Represents if the connection has been established:
+   *  - TCP handshake
+   *  - TLS negotiated
+   *  - mongodb handshake (saslStart, saslContinue), includes authentication
+   * Once connection is established, command logging can log events (if enabled)
+   */
+  public established: boolean;
```

Comment:
What is the expected behavior with command logging when we re-authenticate?

Reply:
Reauthentication will send commands we don't want to log I think.  We may either need to 

- add logic that more specifically determines whether or not a command should be logged
- set `established` to false when reauthenticating and then set it back to true after a successful reauthentication

Reply:
Sorry I didn't notice this earlier ( I didn't actually check ) but instead of modifying `established` when we reauthenticate, could we just check the `context.reauthenticating` field?

one idea is a getter on the connection class (or we could just inline it):
```typescript
get shouldEmitAndLog(): boolean {
  return this.monitorCommands ||
      (this.established && !this.context.reauthenticating && this.mongoLogger?.willLog(SeverityLevel.DEBUG, MongoLoggableComponent.COMMAND))
}
```

Reply:
Isn't it more accurate to edit established as well, though? Since during authentication, the connection is not established.

Reply:
I understood established to mean that the connection had been established, not that it was "ready"/currently established.  What I would like to avoid is duplicate data that might get out of sync.  If established=false can also represent a connection reauthenticating, then it's possible for us to end up in a scenario context.reauthenticating=true with established=false (or vice versa) - which is a poorly defined state for the driver.

One way to avoid this is to define established as whether or not the connection was initially established and rely on context.reauthenticating (my suggestion).

Reply:
That sounds good. I've removed the extra auth changes to `conn.established` and am using `this.authContext?.authenticating` in the new getter `shouldEmitAndLogCommand` instead.